### DISCUSSION TITLE
Error while using File Based Voltage Source

### DISCUSSION THREAD

**Discussion by BertheGrandPied** [ra3xdh/qucs_s#613] [Q&A]
I'm trying to use the File Based Voltage Source (on Debian 12), but impossible to make it work. I tried to simplify at max the .sch, see the picture.

![qucs_sch](https://github.com/ra3xdh/qucs_s/assets/47215865/f54d1e01-20c0-41e6-93c5-47ab07c5ad0e)

I have errors "Instance: av1   Message: cannot open file", and my table in the DAT file is not loaded (get message "Warning from checkvalid: vector _net0 is not available or has zero length." that's seems to show that my table is not loaded).

And still get this error, and no table loaded. I tried to change my DAT format (use comma separated, space separated CSV, etc) cant' make it work. Impossible to get it work...

Here is my file (as txt to be uploaded) : [vfile.dat.txt](https://github.com/ra3xdh/qucs_s/files/14397523/vfile.dat.txt)

I can't find documentation about this feature, the things I find about QUCS are not working. Is it a bug or I'm doing wrong ? If so, is there a working example ?

Thanks for any clue on that.

And a big thanks for these software which is really nice to use (it's really an improvement compare to other software I used in the past)

**Comment by tomhajjar** [comment]
Here is an old Qucs example that runs under Qucs-S.

![2024-02-25_211953](https://github.com/ra3xdh/qucs_s/assets/102838604/7e6ff19a-b9f4-4c32-b2ec-789260260732)

[File_Data_prj.zip](https://github.com/ra3xdh/qucs_s/files/14398918/File_Data_prj.zip)

**Comment by ra3xdh** [comment]
Look for file source example also here: https://github.com/ra3xdh/qucs_s/issues/254

**Comment by ra3xdh** [comment]
I have just checked the latest development build and everything works as expected. If nothing above helps then you have probably wrong configuration of XSPICE extensions for Ngspice. If you are using build Ngspice from source this may mean wrong configure options at compile time.

**Comment by ra3xdh** [comment]
I am transferring this issue to discussions: https://github.com/ra3xdh/qucs_s/discussions

**Comment by BertheGrandPied** [comment]
Thanks for all theses answers.

I tried to open the project example, but still the same error when running the simulation :

>  Ngspice started...
> Using SPARSE 1.3 as Direct Linear Solver
> ERROR: fatal error in ngspice, exit(1)
> Note: No compatibility mode selected!
> Circuit: * qucs 24.1.0  /path/to/vfile_test_tran.sch
> Reducing trtol to 1 for xspice 'A' devices
> Doing analysis at TEMP = 27.000000 and TNOM = 27.000000
> Instance: av2   Message: cannot open file /path/to//vfile_test_data.csv
> Simulation finished

And same using the provided vfile by ra3xdh. I'm confused, because that looks simple.

I'm using the ngspice build from the Debian APT package, which is build with XSpice in : https://packages.debian.org/fr/bookworm/ngspice

I tried to use backports to get a more recent version, nothing change.

I'll give a try with my own compile NGspice. If not working, I will give a try to the windows version, see if it's working (I'm using appimage for Linux, try the deb package too)

**Comment by BertheGrandPied** [comment]
I tried my dat file on the windows version, it works. I give a try to understand why it's not working on my debian.

**Comment by BertheGrandPied** [comment]
I tried to build myself Qucs-s with Qt5 and Qt6, still not working. Try to make my own build of ngspice with xspice -> nothing change.

I'm confused, looks like qucs can"t find the path of my file. Don't know what to do to get more infos about what happen (I tried to get a look to the source code in order to add flags or others feedback, but the code is too complex for my knowledge).

**Comment by BertheGrandPied** [comment]
Last update, everythings works in a VM ubuntu 22.04 with the AppImage

So I supposed thaht show a problem with my Debian 12 bookworm.

**Comment by BertheGrandPied** [comment]
I finally find a way to make it works !

If I put my DAT file in my ~/.qucs (an not a sub folder inside this folder) side by side with my .sch file, everything is ok !

It looks like a problem of path handling... Not sure to understand what happen.
Is it normal ?

I can reproduce this behavior in my VM with Ubuntu, if I placed everything in a subfolder, the qucs can't open the DAT file (but show a message with the correct path "can't open file").

**Comment by ra3xdh** [comment]
> I can reproduce this behavior in my VM with Ubuntu, if I placed everything in a subfolder, t

I didn't observe path problem with file source. Qucs-S accepts bot relative and absolute paths on my machine. Try to specify the absolute path in the file source parameters.

**Comment by BertheGrandPied** [comment]
Hello,

I tried absolute and relative path. and finally I understood my problem ; relative path are define from ~/.qucs folder, not relative to the placement of the .sch and that confused me.

Thanks a lot to take time to help me.
